rust/src/utils.rs

```diff
@@ -15,42 +18,100 @@ pub fn is_eligible(signature: &ParticipantTaskSignature, threshold: f64) -> bool
     } else if threshold > 1_f64 {
         true
     } else {
```

Comment:
nit: I would remove one level of indentation here:

```rust
    if threshold < 0_f64 {
        return false;
    } else if threshold > 1_f64 {
        return true;
    }
    // ...
```

Reply:
i'll change that.